src/ResultSet.php

```diff
@@ -21,6 +22,50 @@ final class ResultSet implements IteratorAggregate
     private array $results = [];
     private bool $isValid = true;
 
+    public function getIterator(): ArrayIterator
+    {
+        return new ArrayIterator($this->results);
+    }
+
+    public function isValid(): bool
+    {
+        return $this->isValid;
+    }
+
+    /**
+     * @psalm-return array<string, Error>
+     */
+    public function getErrorObjects(): array
+    {
+        return $this->getAttributeToErrorsMap(static fn (Result $result) => $result->getErrorObjects());
+    }
+
+    /**
+     * @return string[][]
+     * @psalm-return array<string, array<int|string, string>>
+     */
+    public function getErrors(): array
+    {
+        return $this->getAttributeToErrorsMap(static fn (Result $result) => $result->getErrors());
+    }
+
+    public function getErrorsIndexedByPath(string $separator = '.'): array
+    {
+        return $this->getAttributeToErrorsMap(static fn (Result $result) => $result->getErrorsIndexedByPath($separator));
+    }
+
+    private function getAttributeToErrorsMap(Closure $getErrorsClosure): array
```

Comment:
```suggestion
    private function errorsMap(Closure $getErrorsClosure): array
```

Reply:
Yes, it's probably fine to use shorter variation here, but I'd leave `get` prefix.

Reply:
:heavy_check_mark: